### ISSUE TITLE
tsh attempts relogin for "ambiguous host" errors

### ISSUE THREAD

**Issue by nklaassen** [gravitational/teleport#24943]
Expected behavior:

`tsh ssh user@ambiguous-host` should print a helpful message about `ambiguous-host` matching multiple nodes

Current behavior:

`tsh` immediately attempts to re-login after receiving an ambiguous host error
```
$ tsh ssh nklaassen@ambiguous-host
Enter password for Teleport user nic:
```

Actually tsh will attempt relogin for all NotFound errors https://github.com/gravitational/teleport/blob/1f02e4807e24a960088fc3cfa610e2397948f8e6/lib/client/api.go#L570-L573

Bug details:
- Teleport version: v13.0.0-alpha.1 (no issue in 12.1.5)
- Recreation steps: create multiple ssh direct-dial nodes with matching hostnames, attempt `tsh ssh user@hostname`.
- Debug logs

**Comment by r0mant** [comment]
@rosstimothy Using `git bisect` I triaged it to this commit: https://github.com/gravitational/teleport/commit/3113cd26bbaf90046a18fb51e82ad93950f2c55e. Can you please take a look?

**Comment by Joerger** [comment]
> @Joerger it looks like the NotFoundError check was added in #19420. Is it required for some new functionality or can we remove it?

Sorry for the late reply. This check is used to initiate re-login when the user doesn't have a key/cert, such as:
```
ERROR: open /home/bjoerger/.tsh/keys/proxy.example.com/dev-x509.pem: no such file or directory
```

With my changes in that PR, this error ended up being the error that would be propagated and needed to be caught.

**Comment by ravicious** [comment]
Would it be possible to rewrap it in some other error instead of retrying on `NotFoundError`?

Last time I looked at `RetryWithRelogin`, historically we've been retrying on some other generic errors as well, such as `BadParameterError`. It led to some extra accidental complexity, such as `RetryWithRelogin` having a separate case for `utils.IsPredicateError`.

I couldn't find info on why we'd retry on `BadParameterError` but now that I've read your comment, I think the story must've been similar to `NotFoundError`.

**Comment by Joerger** [comment]
Yes I think you're right. We could add a `NoCredentials` error and return/check that instead.
